### PR TITLE
feat(configurations): Configurations client implementation

### PR DESCRIPTION
**Pull request by bikamani**
This PR does not include:
1. TestQueries and Edge Configurations APIs until we have more information from service team
2. e2e tests

### CODE REVIEW

File: sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        /// </summary>
+        internal ConfigurationsClient(ConfigurationRestClient configurationRestClient)
+        {
+            Argument.AssertNotNull(configurationRestClient, nameof(configurationRestClient));
+
+            _configurationRestClient = configurationRestClient;
+        }
+
+        /// <summary>
+        /// Gets a configuration on the IoT Hub for automatic device/module management
```
Comment (azabbasi) [inline sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs:40]: Gets a twin configuration ?

File: sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The http response <see cref="Response{T}"/>.</returns>
+        public virtual Response DeleteConfiguration(TwinConfiguration configuration, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(configuration, nameof(configuration));
+            string ifMatchHeaderValue = IfMatchPreconditionExtensions.GetIfMatchHeaderValue(precondition, configuration.Etag);
+            return _configurationRestClient.Delete(configuration.Id, ifMatchHeaderValue, cancellationToken);
+        }
+
+        /// <summary>
```
Comment (azabbasi) [inline sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs:121]: if the count is left as null, how many does the service send back ? is there a cap?

Comment (azabbasi) [reply sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs:121]: I think that is worth mentioning in the comments.

File: sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs (hunk 1 above)
Comment (timtay-microsoft) [reply sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs:40]: +1 to this point. If the return type is TwinConfiguration, then the comment should say "twin configuration" not just "configuration"

**Review by timtay-microsoft — APPROVED** [review]
Clarify Azad's question about the list API, but this looks good to me otherwise

File: sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs (hunk 2 above)
Comment (bikamani) [reply sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs:121]: I do not know, I'll add this question to our ongoing thread

File: sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs (hunk 1 above)
Comment (bikamani) [reply sdk/iot/Azure.Iot.Hub.Service/src/ConfigurationsClient.cs:40]: Sure will change. Does the same apply to other APIs?

**Pull request merged**